saleor/payment/models.py

```diff
@@ -101,15 +108,15 @@ def get_authorized_amount(self):
 
         # There is no authorized amount anymore when capture is succeeded
         # since capture can only be made once, even it is a partial capture
-        if self.transactions.filter(
-                kind=TransactionKind.CAPTURE, is_success=True).exists():
+        if any([txn.kind == TransactionKind.CAPTURE
+                and txn.is_success for txn in self.cached_transactions]):
```

Comment:
We do it a bit differently in Saleor. At this point, we should assume that the transactions are prefetched and we can safely iterate over `self.transactions.all()`. E.g. I've checked that `get_authorized_amount` is only used in `resolve_total_authorized` (indirectly). We have an optimizer there that already prefetches payments:

```
@gql_optimizer.resolver_hints(prefetch_related='payments')
```

This is probably not enough and we should also prefetch transactions, something like this:
```
@gql_optimizer.resolver_hints(prefetch_related='payments__transactions')
```

Maybe an easier example is with `can_capture` - this function is used by `order. can_capture` which in turn is used for example by `dashboard:order-details` view. It means that this view should prefetch transactions [here](https://github.com/mirumee/saleor/blob/master/saleor/dashboard/order/views.py#L119).

If you look around the codebase, we don't use Django signals at all and this is on purpose, as we find them difficult to debug and error-prone. I'd change your implementation to only iterate over `self.transactions.all()`. I'm also fine with adding the prefetches in a separate PR.

Reply:
I see I see, thanks a lot for the explanation. I'm just curious why there is no signals at all in Saleor and that's why. 😄

Reply:
I've updated the PR and going to submit another PR later to support prefetch for transactions